x/nft/base/handler.go

```diff
@@ -0,0 +1,161 @@
+package base
+
+import (
+	"fmt"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/x"
+	"github.com/iov-one/weave/x/nft"
+	"github.com/iov-one/weave/x/nft/blockchain"
+	"github.com/iov-one/weave/x/nft/ticker"
+	"github.com/iov-one/weave/x/nft/username"
+)
+
+const (
+	updateApprovalCost = 0
+)
+
+// RegisterRoutes will instantiate and register all handlers in this package
+func RegisterRoutes(r weave.Registry, auth x.Authenticator, issuer weave.Address) {
+	//TODO: might want to move that to init, this looks like an overhead we can avoid
+	bucketMap := map[string]Bucket{}
+	bucketMap[nft.Type_Username.String()] = Bucket{username.NewBucket().Bucket}
+	bucketMap[nft.Type_Ticker.String()] = Bucket{ticker.NewBucket().Bucket}
+	bucketMap[nft.Type_Blockchain.String()] = Bucket{blockchain.NewBucket().Bucket}
+
+	for _, v := range nft.Type_name {
+		if _, ok := bucketMap[v]; !ok {
+			panic(fmt.Sprintf("Bucket not registered in base nft handler: %s", v))
+		}
+	}
+
+	handler := NewApprovalOpsHandler(auth, issuer, bucketMap)
+	r.Handle(nft.PathAddApprovalMsg, handler)
+	r.Handle(nft.PathRemoveApprovalMsg, handler)
+}
+
+type Bucket struct {
```

Comment:
pp: when you replace this by an interface that has Get/Save you don't need a struct type and you can act on the passed buckets

Reply:
I might need to wrap buckets map into a type or something, yes